### PR TITLE
feat : Video Listing Page

### PR DESCRIPTION
**Pull request by vikrantkode**
### Routes added

https://user-images.githubusercontent.com/54833758/161463581-333cc0f8-35c7-43a3-9116-e7a9a8d782eb.mp4

note : if preview link not works

### CODE REVIEW

File: my-app/src/components/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "./Sidebar.css";
+
+const Sidebar = () => {
+  return (
+    // <div className="container">
```
Comment (bhavyaj12) [inline my-app/src/components/Sidebar/Sidebar.jsx]: Removed unused commented code

File: my-app/src/components/VideoCard/ViCard.css [hunk 2]
```diff
@@ -31,8 +22,8 @@
     border-top-right-radius: 0.5rem;
 }
 .title {
-    margin-left: 2rem;
-    margin-top: 1rem;
+    margin-left: 0.5rem;
+    /* margin-top: 0.5rem; */
```
Comment (bhavyaj12) [inline my-app/src/components/VideoCard/ViCard.css]: Remove commented css

File: my-app/src/components/VideoCard/ViCard.css [hunk 3]
```diff
@@ -46,49 +37,37 @@
 }
 
 .content{
-    margin-left: 2rem;
+    margin-left: 1rem;
```
Comment (bhavyaj12) [inline my-app/src/components/VideoCard/ViCard.css]: Use shorthand

File: my-app/src/components/VideoCard/ViCard.css [hunk 4]
```diff
@@ -46,49 +37,37 @@
...
     justify-content: space-around;  
     border: none;
-    flex-wrap: wrap;
+    align-items: center;
 }
-.footer__buttons{
+.footer_info{
     display: flex;
     justify-content: space-around;
     margin-left:auto ;
```
Comment (bhavyaj12) [inline my-app/src/components/VideoCard/ViCard.css]: Code format and space after : is not consistent

File: my-app/src/index.js [hunk 5]
```diff
@@ -3,14 +3,17 @@ import ReactDOM from "react-dom";
 import App from "./App";
 import { makeServer } from "./server";
 import {BrowserRouter} from "react-router-dom"
```
Comment (bhavyaj12) [inline my-app/src/index.js]: Formatting should be consistent for { }

File: my-app/src/pages/VideoListing/VideoListing.css [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    margin-left: 1rem;
+  }
+  
+  .parent{ 
+    grid-area: main;
+    gap: 3rem;
+    width: 100%;    
+    display: flex;
+    flex-wrap: wrap;
+    /* justify-content: center ; */
```
Comment (bhavyaj12) [inline my-app/src/pages/VideoListing/VideoListing.css]: Removed commented css

File: my-app/src/pages/VideoListing/VideoListing.jsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import "./VideoListing.css"
+import { useVideoList } from '../../context/Video-Context'
+const VideoListing = () => {
+    const {videoList} = useVideoList();
+  return (
+    <div className='container_videolist '>
+        <Nav/>
+        <Sidebar />
+        <div className="parent">
+            {/* <div className='card_top'>    */}
```
Comment (bhavyaj12) [inline my-app/src/pages/VideoListing/VideoListing.jsx]: Remove commented JavaScript

**Review by bhavyaj12 — APPROVED** [review]
Preview link is blank, correct it
Code is easy to understand, good work.
Have left a few minor corrections.

File: my-app/src/components/Sidebar/Sidebar.jsx (hunk 1 above)
Comment (vikrantkode) [reply my-app/src/components/Sidebar/Sidebar.jsx]: i just deleted commented code

File: my-app/src/components/VideoCard/ViCard.css (hunk 3 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: okay

File: my-app/src/components/VideoCard/ViCard.css (hunk 4 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: did the formatting

File: my-app/src/pages/VideoListing/VideoListing.css (hunk 6 above)
Comment (vikrantkode) [reply my-app/src/pages/VideoListing/VideoListing.css]: removed

File: my-app/src/index.js (hunk 5 above)
Comment (vikrantkode) [reply my-app/src/index.js]: done formatting

File: my-app/src/pages/VideoListing/VideoListing.jsx (hunk 7 above)
Comment (vikrantkode) [reply my-app/src/pages/VideoListing/VideoListing.jsx]: removed

File: my-app/src/context/Video-Context.jsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        try {
+            const resp = await axios.get(`/api/videos`)
+            console.log(resp.data.videos)
+            setVideoList(resp.data.videos)
+        } catch (error) {
+            console.log(error)
+        }
+    }
+
+    useEffect(() => {
```
Comment (harshitpaliwal95) [inline my-app/src/context/Video-Context.jsx]: Good use of useeffect here

**Review by harshitpaliwal95 — APPROVED** [review]
- everything looks good 
- ui looks clean 
Good work vikhrat you can merge

**Pull request merged**